scout/adapter/mongo/query.py

```diff
@@ -206,86 +206,80 @@ def build_query(
         # set up the fundamental query params: case_id, category, type and
         # restrict to list of variants (if var list is provided)
         for criterion in FUNDAMENTAL_CRITERIA:
-            if criterion == "case_id":
```

Comment:
For the most part just switching (pun intended) to match-case, and removing some debug statements. Maybe a table of function calls would be even better, but one thing at a time.